.github/workflows/test-pr.yaml

```diff
@@ -19,6 +19,7 @@ jobs:
     needs: [detect-changes]
     runs-on: ubuntu-latest
     continue-on-error: true
+    if: ${{ needs.detect-changes.outputs.templates != '[]' && needs.detect-changes.outputs.templates != ''}    
```

Comment:
This is currently working as expected by this PR and the tests are only run on the Templates that were updated. Hence, this PR (which does not update Templates) does not run any tests.

Also, see another example https://github.com/devcontainers/template-starter/pull/31 only updated `color` Template, hence, only tests against `color` were executed.

Reply:
good point. this PR is not changing any template, it not supposed to. mostly did this pull request, because I did run into that issue, posted above and I found out how to fix it. I am trying to help my company with devcontainers and I know we/they could run into that error. So this statement would help whomever is learning this process like I to not have to waste energy on non-issues. 

but all good, we can close it out if it's not needed. Thank you for your time to review. love the work you do.

Reply:
@Cog-Smith Is it possible to point me to your workflow run that failed? I am curious to learn why it's failing for your particular scenario, but always passed for other repos/PRs.

Happy to merge this PR if we think it's needed, thanks!

Reply:
@samruddhikhandale I faced exactly the same problem today, and I fixed in my repo in a similar way, before finding this PR.

I just cloned the template repo and added a new template to the `src` folder. https://github.com/brpaz/devcontainer-templates/actions/runs/10771852695

I also had to add `actions/checkout` to clone the repo. 

I am not sure how this can be working for other repos. Unless matrix strategy already support, ignoring a job when receiving an empty list.